src/export.rs

```diff
@@ -86,61 +84,28 @@ impl<T> MaybeUninit<T> {
         self.inner.as_mut_ptr()
     }
 
+    #[cfg(feature = "nightly")]
     pub fn write(&mut self, value: T) -> &mut T {
         self.inner.write(value)
     }
-}
-
-#[cfg(not(feature = "nightly"))]
-pub struct MaybeUninit<T> {
-    value: Option<T>,
-}
-
-#[cfg(not(feature = "nightly"))]
-const MSG: &str =
-    "you have hit a bug (UB) in RTFM implementation; try enabling this crate 'nightly' feature";
-
-#[cfg(not(feature = "nightly"))]
-impl<T> MaybeUninit<T> {
-    pub const fn uninit() -> Self {
-        MaybeUninit { value: None }
-    }
-
-    pub fn as_ptr(&self) -> *const T {
-        if let Some(x) = self.value.as_ref() {
-            x
-        } else {
-            unreachable!(MSG)
-        }
-    }
-
-    pub fn as_mut_ptr(&mut self) -> *mut T {
-        if let Some(x) = self.value.as_mut() {
-            x
-        } else {
-            unreachable!(MSG)
-        }
-    }
 
+    #[cfg(not(feature = "nightly"))]
     pub unsafe fn get_ref(&self) -> &T {
-        if let Some(x) = self.value.as_ref() {
-            x
-        } else {
-            unreachable!(MSG)
-        }
+        &*self.inner.as_ptr()
     }
 
+    #[cfg(not(feature = "nightly"))]
     pub unsafe fn get_mut(&mut self) -> &mut T {
-        if let Some(x) = self.value.as_mut() {
-            x
-        } else {
-            unreachable!(MSG)
-        }
+        &mut *self.inner.as_mut_ptr()
```

Comment:
As this is the straightforward implementation, maybe we can remove the cfg and implementing it even in nightly?

Reply:
This we can do, I will remove the !nightly flag